tests/test_analysis.py

```diff
@@ -420,50 +420,68 @@ def test_mix_scenarios():
     }
 
 
-def test_scenario_switcher(seed):
+def test_improved_healthsystem_and_care_seeking_scenario_switcher(seed):
     """Check the `ScenarioSwitcher` module can update parameter values in a manner similar to them being changed
-    directly after registration in the simulation (as would be done by the Scenario class)."""
+    directly after registration in the simulation (as would be done by the Scenario class).
+    And check that these parameters can be changed mid-way through the simulation."""
 
     sim = Simulation(start_date=Date(2010, 1, 1), seed=seed)
     sim.register(*(
-        fullmodel(resourcefilepath=resourcefilepath) + [ScenarioSwitcher(resourcefilepath=resourcefilepath)]
+        fullmodel(resourcefilepath=resourcefilepath) +
+        [ImprovedHealthSystemAndCareSeekingScenarioSwitcher(resourcefilepath=resourcefilepath)]
     ))
 
-    # Check that the 'ScenarioSwitcher` is the first registered module.
-    assert 'ScenarioSwitcher' == list(sim.modules.keys())[0]
+    # Check that the 'ImprovedHealthSystemAndCareSeekingScenarioSwitcher` is the first registered module.
+    assert 'ImprovedHealthSystemAndCareSeekingScenarioSwitcher' == list(sim.modules.keys())[0]
+    module = sim.modules['ImprovedHealthSystemAndCareSeekingScenarioSwitcher']
 
     # Change the parameters for max_healthsystem_function and max_healthcare_seeking via the ScenarioSwitcher
-    sim.modules['ScenarioSwitcher'].parameters['max_healthsystem_function'] = True
-    sim.modules['ScenarioSwitcher'].parameters['max_healthcare_seeking'] = True
+    max_healthsystem_function = [False, True]
+    max_healthcare_seeking = [False, True]
+    year_of_change = sim.date.year + 1
+
+    module.parameters['year_of_switch'] = year_of_change
+    module.parameters['max_healthsystem_function'] = max_healthsystem_function
+    module.parameters['max_healthcare_seeking'] = max_healthcare_seeking
 
     # Initialise the population
     sim.make_initial_population(n=100)
 
-    # Check that all the parameter values in the simulation are updated to be the value expected.
-    updated_values = get_parameters_for_improved_healthsystem_and_healthcare_seeking(
-        resourcefilepath=resourcefilepath,
-        max_healthsystem_function=True,
-        max_healthcare_seeking=True
-    )
+    def check_parameters(sim, phase_of_simulation) -> None:
+        """Check that the paramter values in the simulation currently match expectation of that phase of the
+        simulation (0=first phase before the change, 1=second phase after the change)."""
+
+        updated_values = get_parameters_for_improved_healthsystem_and_healthcare_seeking(
+            resourcefilepath=resourcefilepath,
+            max_healthsystem_function=max_healthsystem_function[phase_of_simulation],
+            max_healthcare_seeking=max_healthcare_seeking[phase_of_simulation]
+        )
+
+        for module, param in updated_values.items():
```

Comment:
Can't we use the below lines to test `update_parameters` method defined in scenario switcher? I understand it is also calling same `get_parameters_for_improved_healthsystem_and_healthcare_seeking` and I feel like that will give us a clear picture of parameters if they are really changing during simulation.